EventFilter/GEMRawToDigi/interface/VFATdata.h

```diff
@@ -0,0 +1,210 @@
+#ifndef EventFilter_GEMRawToDigi_VFATdata_h
+#define EventFilter_GEMRawToDigi_VFATdata_h
+#include <vector>
+
+namespace gem {
+  class VFATdata 
+  {
+  public:    
+    VFATdata(){}
+    ~VFATdata(){}
+  VFATdata(const uint8_t &b1010_, 
+	   const uint16_t &BC_,
+	   const uint8_t &b1100_, 
+	   const uint8_t &EC_,
+	   const uint8_t &Flag_, 
+	   const uint8_t &b1110_, 
+	   const uint16_t &ChipID_, 
+	   const uint64_t &lsData_, 
+	   const uint64_t &msData_, 
+	   const uint16_t &crc_,
+	   const uint16_t &crc_calc_,
+	   const int &SlotNumber_,
+	   const bool &isBlockGood_) : 
+    fb1010(b1010_),
+      fBC(BC_),
+      fb1100(b1100_),
+      fEC(EC_),
+      fFlag(Flag_),
+      fb1110(b1110_),
+      fChipID(ChipID_),
+      flsData(lsData_),
+      fmsData(msData_),
+      fcrc(crc_),
+      fcrc_calc(crc_calc_),
+      fSlotNumber(SlotNumber_),
+      fisBlockGood(isBlockGood_){
+	fcrc = this->checkCRC();
+      }
+
+    //!Read first word from the block.
+    void read_fw(uint64_t word)
+    {
+      fb1010 = 0x0f & (word >> 60);
+      fBC = 0x0fff & (word >> 48);
+      fb1100 = 0x0f & (word >> 44);
+      fEC = word >> 36;
+      fFlag = 0x0f & (word >> 32);
+      fb1110 = 0x0f & (word >> 28);
+      fChipID = 0x0fff & (word >> 16);
+      fmsData = 0xffff000000000000 & (word << 48);
+    }
+    uint64_t get_fw()
+    {
+      return
+	(static_cast<uint64_t>(fb1010 & 0x0f) <<  60) |
+	(static_cast<uint64_t>(fBC & 0x0fff) <<  48) |
+	(static_cast<uint64_t>(fb1100 & 0x0f) <<  44) |
+	(static_cast<uint64_t>(fEC) <<  36) |
+	(static_cast<uint64_t>(fFlag & 0x0f) <<  32) |
+	(static_cast<uint64_t>(fb1110 & 0x0f) <<  28) |
+	(static_cast<uint64_t>(fChipID & 0x0fff) <<  16) |
+	(static_cast<uint64_t>(fmsData & 0xffff000000000000) >> 48);
+    }
+
+    //!Read second word from the block.
+    void read_sw(uint64_t word)
+    {
+      fmsData = fmsData | (0x0000ffffffffffff & word >> 16);
+      flsData = 0xffff000000000000 & (word << 48);
+    }
+    uint64_t get_sw()
+    {
+      return
+	(static_cast<uint64_t>(fmsData & 0x0000ffffffffffff) <<  16) |
+	(static_cast<uint64_t>(flsData & 0xffff000000000000) >>  48);
+    }
+    
+    //!Read third word from the block.
+    void read_tw(uint64_t word)
+    {
+      flsData = flsData | (0x0000ffffffffffff & word >> 16);
+      fcrc = word;
+    }
+    // make write_word function
+    uint64_t get_tw()
+    {
+      return
+	(static_cast<uint64_t>(flsData & 0x0000ffffffffffff) <<  16) |
+	(static_cast<uint64_t>(fcrc));
+    }
+        
+    uint8_t   b1010      (){ return fb1010;      }
+    uint16_t  BC         (){ return fBC;         }
```

Comment:
Even here, CMS code rules require to start method names with lower case.
To rename all this could be a bit tricky and i won't insist on it (it wouldn't have been tricky would you have named all class members with m_*, as in GEBdata.h, and in that case f* could have been the renamed method name)